utils/mgmt_system.py

```diff
@@ -1200,7 +1200,14 @@ def info(self):
 
     def list_vm(self):
         """Returns a list from instance IDs currently active on EC2 (not terminated)"""
-        return [inst.id for inst in self._get_all_instances() if inst.state != 'terminated']
+        instances = [inst for inst in self._get_all_instances() if inst.state != 'terminated']
+        result = []
+        for instance in instances:
+            if instance.tags and "Name" in instance.tags and instance.tags["Name"]:
+                result.append(instance.tags["Name"])
+            else:
+                result.append(instance.id)
+        return result
```

Comment:
Unless there's a bug in boto, instance.tags should always be a dict, or dict-like object (like boto.ec2.tags.TagSet).

```python
instances = [inst for inst in self._get_all_instances() if inst.state != 'terminated']
return [i.tags.get('Name', i.id) for i in instances]
```